python/cudf/cudf/core/dataframe.py

```diff
@@ -6239,6 +6240,146 @@ def interleave_columns(self):
             {None: libcudf.reshape.interleave_columns([*self._columns])}
         )
 
+    @_cudf_nvtx_annotate
+    def eval(self, expr: "str", inplace: "bool" = False, **kwargs):
```

Comment:
how worthwhile is it to match the pandas signature here? It seems like it supports some bells and whistles that I am guessing we don't and won't in the near term at least.